app/models/ability.rb

```diff
@@ -0,0 +1,8 @@
+class Ability
+  include CanCan::Ability
+  def initialize(user)
+    can :add_new_food, Recipe do |recipe|
+      recipe.user == user
+    end
+  end
```

Comment:
The above code should be:

```ruby
def initialize(user)
  user ||= User.new # In case no user is logged in

  can :manage, [Food], author_id: user_id
end
```

This will allow all users to `:create` a food entry and `:show`, `:update`, `:destroy` any entries belonging to them.